test/contract-access-controller-integration.test.js

```diff
@@ -0,0 +1,226 @@
+const ipfsPath1 = './orbitdb/tests/contract-access-controller-integration/1/ipfs'
+const ipfsPath2 = './orbitdb/tests/contract-access-controller-integration/2/ipfs'
+
+Object.keys(testAPIs).forEach(API => {
+  describe('orbit-db - ContractAccessController Integration', function () {
+    this.timeout(config.timeout)
+
+    let ipfsd1, ipfsd2, ipfs1, ipfs2, id1, id2
+    let orbitdb1, orbitdb2
+    let web3, accounts, contract
+
+    before(async () => {
+      config.daemon1.repo = ipfsPath1
+      config.daemon2.repo = ipfsPath2
+      rmrf.sync(config.daemon1.repo)
+      rmrf.sync(config.daemon2.repo)
+      rmrf.sync(dbPath1)
+      rmrf.sync(dbPath2)
+      ipfsd1 = await startIpfs(API, config.daemon1)
+      ipfsd2 = await startIpfs(API, config.daemon2)
+      ipfs1 = ipfsd1.api
+      ipfs2 = ipfsd2.api
+
+      const keystore1 = Keystore.create(dbPath1 + '/keys')
+      const keystore2 = Keystore.create(dbPath2 + '/keys')
+
+      const wallet1 = await open({ privateKey: '0x3141592653589793238462643383279502884197169399375105820974944592' })
+      const wallet2 = await open({ privateKey: '0x2141592653589793238462643383279502884197169399375105820974944592' })
+
+      const signer1 = async (id, data) => await wallet1.signMessage({ message: data })
+      const signer2 = async (id, data) => await wallet2.signMessage({ message: data })
+
+      id1 = await IdentityProvider.createIdentity(keystore1, wallet1.address, { type: 'ethers', identitySignerFn: signer1 })
+      id2 = await IdentityProvider.createIdentity(keystore2, wallet2.address, { type: 'ethers', identitySignerFn: signer2 })
+
+      web3 = new Web3(ganache.provider())
+      accounts = await web3.eth.getAccounts()
+      contract = await new web3.eth.Contract(abi)
+                          .deploy({ data: bytecode })
+                          .send({ from: accounts[0], gas: '1000000' })
+
+      AccessControllers.addAccessController({
+        AccessController: ContractAccessController,
+      })
+
+      orbitdb1 = await OrbitDB.createInstance(ipfs1, {
+        ACFactory: AccessControllers,
+        directory: dbPath1,
+        identity: id1
+      })
+
+      orbitdb2 = await OrbitDB.createInstance(ipfs2, {
+        ACFactory: AccessControllers,
+        directory: dbPath2,
+        identity: id2
+      })
+    })
+
+    after(async () => {
+      if (orbitdb1) {
+        await orbitdb1.stop()
+      }
+
+      if (orbitdb2) {
+        await orbitdb2.stop()
+      }
+
+      if (ipfsd1) {
+        await stopIpfs(ipfsd1)
+      }
+
+      if (ipfsd2) {
+        await stopIpfs(ipfsd2)
+      }
+    })
+
+    describe('OrbitDB Integration', function () {
+      let db, db2
+      let dag
+      let dbManifest, acManifest
+
+      before(async () => {
+        db = await orbitdb1.feed('AABB', {
+          identity: id1,
+          accessController: {
+            type: 'eth-contract',
+            web3: web3,
```

Comment:
These params (web3, abi and address) could be passed in at `addAccessController()` above.

Reply:
They need to be passed in on db creation, a peer wont need to pass in abi or contract address as those will be written to the acManifest but the web3 instance and defaultAccount are specific to each db and aren't in the acManifest. addAccessController currently just adds the correct class for getHandlerFor to return and doesnt set any options.

Reply:
Do I understand correctly, abi and contract address need to be given at creation time, web3 and defaultAccount per orbit-db instance? (or per database instance?)